index.bs

```diff
@@ -315,6 +315,30 @@ For example,
 is more readable than
 <code highlight="js">window.scrollBy(50, 0)</code>.
 
+<h3 id="promises">Use Promises</h3>
+
+Asynchronous APIs should generally be designed using promises
+rather than callback functions,
+because APIs using promises are generally easier to use
+than APIs using callback functions.
```

Comment:
IMO the more important reason is because this is the pattern we've settled on for the web platform as a whole. Consistency is important, and you can compose more easily with other web platform APIs (e.g. creating promise chains).